swap/src/cli/trace.rs

```diff
@@ -0,0 +1,147 @@
+use crate::cli::command::Command;
+use anyhow::Result;
+use std::path::Path;
+use tracing::level_filters::LevelFilter;
+use tracing::subscriber::set_global_default;
+use tracing::{Event, Subscriber};
+use tracing_appender::non_blocking::WorkerGuard;
+use tracing_subscriber::fmt::format::{DefaultFields, Format};
+use tracing_subscriber::fmt::time::ChronoLocal;
+use tracing_subscriber::layer::{Context, SubscriberExt};
+use tracing_subscriber::registry::LookupSpan;
+use tracing_subscriber::{fmt, Layer, Registry};
+use uuid::Uuid;
+
+pub fn init_tracing(
+    debug: bool,
+    dir: impl AsRef<Path>,
+    command: &Command,
+) -> Result<Option<WorkerGuard>> {
+    let is_terminal = atty::is(atty::Stream::Stderr);
+
+    let file_logger = |&swap_id| {
+        let appender = tracing_appender::rolling::never(dir, format!("{}.log", swap_id));
+        let (non_blocking, guard) = tracing_appender::non_blocking(appender);
+        (
+            fmt::layer()
+                .with_ansi(is_terminal)
+                .with_target(false)
+                .with_writer(non_blocking),
+            guard,
```

Comment:
FYI: To avoid having to return this `guard`, you can also use `std::mem::forget`. Given that we are not a long-running deamon, this won't cause a harmful memory-leak as we will terminate eventually.

That should simplify some of the APIs.